VaccineCard/VaccineCard/Services/CodeValidationService.swift

```diff
@@ -128,4 +128,22 @@ class CodeValidationService {
         let payload = parts[1]
         return payload.base64DecodedData()
     }
+    
+    fileprivate func getJWSHeader(from  string: String) -> QRHeader? {
+        let parts = string.components(separatedBy: ".")
+        guard parts.count == 3 else {
+            print("Invalid Compact JWS: must have 3 base64 components separated by a dot")
+            return nil
+        }
+        let header = parts[0]
```

Comment:
I assume there would be a failure before getting to this point if parts was empty, but the only thing I'd say here would be to safely check if parts is empty before accessing it with index 0, just to avoid potential index out of range crash - up to your discretion though, obviously don't want to overkill it with safety checks.